### PR TITLE
Skip parsing annotations containing dashes, such as `@Foo-bar`, or `@Foo-`

### PR DESCRIPTION
**Pull request by Ocramius**
Most likely fixes https://github.com/doctrine/collections/issues/180

### CODE REVIEW

**Comment by greg0ire** [comment]
> Skip parsing annotations containing dashes, such as `@Foo-bar`, or `@Foo`

:confused: 	where is the dash in `Foo`?

**Comment by Ocramius** [comment]
@greg0ire corrected :-)

File: lib/Doctrine/Common/Annotations/DocLexer.php [hunk 1]
```diff
@@ -76,6 +78,17 @@ final class DocLexer extends AbstractLexer
         'null'  => self::T_NULL
     );
 
+    /**
+     * Whether the token next token starts immediately, of if there were
```
Comment (greg0ire) [inline lib/Doctrine/Common/Annotations/DocLexer.php]:
```suggestion
     * Whether the next token starts immediately, of if there were
```

File: tests/Doctrine/Tests/Common/Annotations/DocLexerTest.php [hunk 2]
```diff
@@ -215,4 +215,99 @@ public function testRecognizesDoubleQuotesEscapeSequence()
...
+                [
+                    'value'     => '-56',
+                    'position'  => 7,
+                    'type'      => DocLexer::T_INTEGER,
+                ]
+            ]
+        );
+    }
+
+    /** @return void */
```
Comment (greg0ire) [inline tests/Doctrine/Tests/Common/Annotations/DocLexerTest.php]: Why don't we use a return type hint here?

Comment (Ocramius) [reply tests/Doctrine/Tests/Common/Annotations/DocLexerTest.php]: Good point! I though `1.6` had older PHP version support :)

**Comment by Ocramius** [comment]
Ported to `1.7` via 5cfa972716505d343c1e3fb6f806bdb62604aa82

**Comment by Ocramius** [comment]
Ported to `master` via 6403955bada39e7483be380039c01fae2c177848

**Pull request merged**